cask/src/cask/internal/NewThreadPerTaskExecutor.scala

```diff
@@ -0,0 +1,14 @@
+package cask.internal
+
+import java.util.concurrent.{Executor, ThreadFactory}
+
+private[cask] final class NewThreadPerTaskExecutor(val threadFactory: ThreadFactory)
```

Comment:
Do we need to define this explicitly, or should we use `Executors.newVirtualThreadPerTaskExecutor` like https://stackoverflow.com/a/76599122/871202? Maybe using reflection so we can continue to compile against older versions of Java?

Reply:
That can be done with a MethodHandler